### DISCUSSION TITLE
Continuous prediction values for semantic segmentation pixels

### DISCUSSION THREAD

**Discussion by NoaMills-USDA** [azavea/raster-vision#1514] [Q&A] [answered]
I'm running a semantic segmentation model that predicts two classes. I am able to construct a prediction raster of pixels with values 0 and 1. I know that internally, in order for backprop to work, the algorithm creates a prediction vector for each pixel specifying the pseudo-probability that the pixel is each class value (ie [0.3, 0.7] if the algorithm is 70% certain the pixel is class 2). Is it possible to extract this information from the model, instead of just getting binary values for each pixel?

**Comment by lewfish** [comment, marked as answer]
If you pass `smooth_output = True` to `SemanticSegmentationLabelStoreConfig`, you should get the probabilities in the output. See the [docs](https://docs.rastervision.io/en/latest/api_reference/_generated/rastervision.core.data.label_store.semantic_segmentation_label_store_config.SemanticSegmentationLabelStoreConfig.html#rastervision.core.data.label_store.semantic_segmentation_label_store_config.SemanticSegmentationLabelStoreConfig.smooth_output)
